source/agora/test/ValidatorRecurringEnrollment.d

```diff
@@ -131,12 +131,19 @@ unittest
     auto blocks = node_0.getBlocksFrom(0, 2);
     assert(blocks.length == 1);
 
-    network.generateBlocks(Height(GenesisValidatorCycle + 2), true);
-    blocks = node_0.getBlocksFrom(10, GenesisValidatorCycle + 3);
-    assert(blocks[$ - 1].header.height == Height(GenesisValidatorCycle + 2));
-    auto last_enrolls = blocks.retro.take(3).map!(block => block.header.enrollments.length);
-    assert(last_enrolls.sum() == 6);
-    assert(!last_enrolls.any!(count => count > 3));
+    network.generateBlocks(Height(GenesisValidatorCycle), true);
+    auto block20 = node_0.getBlock(Height(GenesisValidatorCycle));
+    assert(block20.header.enrollments.length == 3);
+
+    // This block has only half of the validators enrolled
+    network.generateBlocks(iota(3), Height(GenesisValidatorCycle + 1), true);
```

Comment:
Shouldn't all the nodes be able to generate blocks regardless of the number of validators?

Reply:
In this test only the first 3 enrollments were put into the block so only 3 validators will be active in that block. The other 3 will fetch the block during catchup. The `generateBlocks` calls `addBlock` which assumes enrollments are in the last block of the cycle. In this test that is not the case.

Reply:
I see what you mean. As far as I know, the last half of validators who are not re-enrolled yet can not have the pre-images of current validators at the height of 20.